optimum/habana/transformers/trainer.py

```diff
@@ -450,6 +450,9 @@ def _tune_save_checkpoint(self, checkpoint_dir: str):
         output_dir = os.path.join(checkpoint_dir, f"{PREFIX_CHECKPOINT_DIR}-{self.state.global_step}")
         self.save_model(output_dir, _internal_call=True)
         if self.args.should_save:
+            # TODO
+            # Update the `TrainerControl` state to where we are currently
+            # self.state.stateful_callbacks["TrainerControl"] = self.control.state()
```

Comment:
I had missed that change! Did you comment it because it doesn't work?

Reply:
I added a TODO comment because it works with the new accelerator and we are on 0.33.0. However, I haven't tested or integrated it yet. We should definitely transition to using it.

Reply:
Sounds good :+1: 
We can wait for native Gaudi support in Accelerate which should be merged soon: https://github.com/huggingface/accelerate/pull/3378/